### DISCUSSION TITLE
TuringCourses/major/database/

### DISCUSSION THREAD

**Discussion by giscus** [ZJU-Turing/TuringCourses#76] [General]
# TuringCourses/major/database/

图灵班学习指南

https://zju-turing.github.io/TuringCourses/major/database/

**Comment by DaViD-Pigeon** [comment]
今年高云君老师的数据库老师和助教一起发疯，最后一节复习课全用来发疯了（？），助教在minisql的验收上也保持了一贯的离谱作风，高老师班最大的卖点——平时分高在今年也完全没有体现。如果能选别的老师还是快逃比较好。

**Comment by HobbitQia** [comment]
依稀记得某次实验课去找孙老师的助教 njzgg，他说目前测试还不是很完善，之前测试的方法是把 CMU 15-445 的测试代码爬了下来，以后应该要做我们自己的评测平台，类似于 gradescope 的方式。
最后，他手指向天空，笑着说：孙老师说了，我们要全面对标 CMU，起码这门课是这样。

毫无疑问这门课的 workload 非常大，但我仍然推荐非必修的同学来选这门课，在 debug 中你能真正掌握数据库的核心知识，何况还有如此优秀负责的老师的和助教。
真诚感谢孙建伶老师，以及这两年的助教，应承峻学长和聂俊哲学长。

**Comment by MTDickens** [comment]
大作业的框架，其实已经是 bustub 的超级简化版了。原版的 bustub 里面存在着大量的 r-value reference, move sementics 以及各种模板技巧，但是 zju 魔改框架里面，这些就被替换掉了。所以说，C++ 所需的其实不多，只需要 OOP 的一些基础知识就够用了。

另外 cmake 和 gtest，其实基本上也可以 0 基础先学。你可能用到的，都是最最简单的功能。因此其实有一个基本概念就行。
